web/src/left_sidebar_navigation_area.ts

```diff
@@ -81,28 +81,16 @@ export function update_dom_with_unread_counts(
     }
 }
 
-// TODO: Rewrite how we handle activation of narrows when doing the redesign.
-// We don't want to adjust class for all the buttons when switching narrows.
-
-function remove($elem: JQuery): void {
-    $elem.removeClass("active-filter active-sub-filter");
-}
-
-function deselect_top_left_corner_items(): void {
-    remove($(".top_left_all_messages"));
-    remove($(".top_left_starred_messages"));
-    remove($(".top_left_mentions"));
-    remove($(".top_left_recent_view"));
-    remove($(".top_left_inbox"));
-    remove($(".top_left_my_reactions"));
+function deselect_top_left_corner_items(narrow_to_activate = ""): void {
```

Comment:
The logic seems correct but the function name is misleading. Maybe just `select_top_left_corner_item` and make `narrow_to_activate` a required parameter. So that `select_top_left_corner_item("")` works to deactivate all items and is clear to read too.